### PR TITLE
[IMP] payment,website_payment improvements

### PR DESCRIPTION
**Pull request by bouvyd**
Task: https://www.odoo.com/web#id=19919&view_type=form&model=project.task&menu_id=3942&action=327
Pad: https://pad.odoo.com/p/r.Epds3YvFsbeiplxx

pay.php without the .php part

### CODE REVIEW

File: addons/website_payment/controllers/main.py [hunk 1]
```diff
@@ -27,3 +27,103 @@ def delete(self, delete_pm_id=None):
             pay_meth = request.env['payment.method'].browse(int(delete_pm_id))
             pay_meth.unlink()
         return request.redirect('/my/payment_method')
+
+    @http.route(['/website-payment/<int:tx_id>', '/website-payment'], type='http', auth='public', website=True)
```
Comment (antonylesuisse) [inline addons/website_payment/controllers/main.py]: /website_payment/

File: addons/payment/models/payment_acquirer.py [hunk 2]
```diff
@@ -370,10 +353,14 @@ class PaymentTransaction(osv.Model):
     """
     _name = 'payment.transaction'
     _description = 'Payment Transaction'
-    _inherit = ['mail.thread']
```
Comment (KangOl) [inline addons/payment/models/payment_acquirer.py]: Shouldn't keep it and track some fields for change?

File: addons/payment/models/payment_acquirer.py [hunk 3]
```diff
@@ -370,10 +353,14 @@ class PaymentTransaction(osv.Model):
...
     _order = 'id desc'
     _rec_name = 'reference'
 
+    def _lang_get(self, cr, uid, context=None):
+        lang_ids = self.pool['res.lang'].search(cr, uid, [], context=context)
+        languages = self.pool['res.lang'].browse(cr, uid, lang_ids, context=context)
+        return [(language.code, language.name) for language in languages]
+
     _columns = {
         'date_create': fields.datetime('Creation Date', readonly=True, required=True),
```
Comment (KangOl) [inline addons/payment/models/payment_acquirer.py]: what's wrong with `create_date`?

File: addons/website_payment/controllers/main.py [hunk 4]
```diff
@@ -27,3 +27,103 @@ def delete(self, delete_pm_id=None):
             pay_meth = request.env['payment.method'].browse(int(delete_pm_id))
             pay_meth.unlink()
         return request.redirect('/my/payment_method')
+
+    @http.route(['/website_payment/<int:tx_id>', '/website_payment'], type='http', auth='public', website=True)
+    def payment(self, tx_id=None, reference='', amount=False, currency_id=None, country_id=None, partner_name='Odoo', partner_id=None, acquirer_id=None, **kw):
+        error_messages = []
+        tx_obj = request.env['payment.transaction']
+        default_country = request.env['res.company'].search([], limit=1).country_id
```
Comment (KangOl) [inline addons/website_payment/controllers/main.py]: Should use company of user: ```request.env.user.company_id```

File: addons/website_payment/models/payment_transaction.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from openerp import api, fields, models
+
+
+class PaymentTransaction(models.Model):
+    _description = 'Payment Transaction'
+    _inherit = 'payment.transaction'
+
+    # let's set a default on the provider
+    acquirer_id = fields.Many2one('payment.acquirer', default=lambda s: s.env['ir.values'].get_default('payment.acquirer', 'acquirer_id'))
```
Comment (KangOl) [inline addons/website_payment/models/payment_transaction.py]: Not sure about the default value.
You should not query `ir.values` yourself.

File: addons/website_payment/models/res_config.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+                                       string="Default Acquirer",
+                                       help="Default payment acquirer for website payments; your provider needs to be visible in the website.",
+                                       domain="[('website_published','=',True)]"
+                                       )
+
+    @api.model
+    def get_default_acquirer(self, fields):
+        default_acquirer = False
+        if 'default_acquirer' in fields:
+            default_acquirer = self.env['ir.values'].get_default('payment.acquirer', 'acquirer_id')
```
Comment (KangOl) [inline addons/website_payment/models/res_config.py]: should be 
```python
self.env['ir.values'].get_default('payment.transaction', 'acquirer_id')
```

**Pull request merged**